ably/rest/auth.py

```diff
@@ -170,79 +179,54 @@ def create_token_request(self, key_name=None, key_secret=None,
         if query_time is None:
             query_time = self.auth_options.query_time
 
-        if not token_params.get("timestamp"):
+        if not timestamp:
             if query_time:
-                token_params["timestamp"] = self.ably.time()
+                timestamp = self.ably.time()
             else:
-                token_params["timestamp"] = self._timestamp()
+                timestamp = self._timestamp()
+
+        token_request["timestamp"] = int(timestamp)
 
-        token_params["timestamp"] = int(token_params["timestamp"])
+        token_request['ttl'] = ttl or TokenDetails.DEFAULTS['ttl'] * 1000
```

Comment:
Yup, I did the same in Ruby, I think this is right as it's more idiomatic for Python